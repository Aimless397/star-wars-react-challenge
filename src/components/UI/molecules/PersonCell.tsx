import { CharacterContext } from "../../../context/CharacterContext";
import { ICharacter } from "../../../interfaces/interfaces";
import { useContext } from "react";

interface Props {
  data: ICharacter[];
}

export const PersonCell = ({ data }: Props) => {
  const { selectCharacter, characterSelected } = useContext(CharacterContext);

  const changeSelectedCharacter = (name: string) => {
    if (characterSelected?.name !== name) {
      selectCharacter({ name });
    }
  };

  return (
    <>
      {data.map((result) => (
        <div
          key={result.name}
          onClick={() => changeSelectedCharacter(result.name)}
          data-testid="personCell"
          className={`personCell ${
            characterSelected?.name === result.name ? "personCellHover" : ""
          }`}
        >
          <div>
            <div>
              <span className="characterNameSpan">{result.name}</span>
              <br />
              <span className="characterSpeciesSpan">
                {result.species.length ? result.species : "Human"} from{" "}
                {result.homeworld}
              </span>
            </div>
          </div>
          <div className="my-auto">
            <i className="gg-chevron-right"></i>
          </div>
        </div>
      ))}
    </>
  );
};
