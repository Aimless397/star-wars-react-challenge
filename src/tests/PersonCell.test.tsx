import { describe, expect, test } from "@jest/globals";
import { render } from "@testing-library/react";
import { PersonCell } from "../components/UI/molecules/PersonCell";

describe("<PersonCell />", () => {
  test("should be rendered", () => {
    const { container } = render(<PersonCell data={[]} />);

    expect(container).toMatchSnapshot();
  });
});
