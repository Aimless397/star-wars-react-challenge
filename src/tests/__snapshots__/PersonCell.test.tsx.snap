// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonCell /> should be rendered 1`] = `<div />`;
